.github/actions/build-push-artifacts/action.yml

```diff
@@ -0,0 +1,63 @@
+name: 'Builds and Push Artifacts'
+description: 'Builds and pushes a Docker image, S3 env vars file, and dispatches a Twistlock workflow'
+inputs:
+  environment:
+    required: true
+    default: "dev"
+    type: string
+  ref:
+    description: 'Git reference (commit SHA, branch, or tag) to build'
+    required: true
+  aws-access-key-id:
+    description: 'AWS Access Key ID'
+    required: true
+  aws-secret-access-key:
+    description: 'AWS Secret Access Key'
+    required: true
+  role-to-assume:
+    description: 'AWS role to assume'
+    required: true
+
+runs:
+  using: "composite"
+  steps:
+    - name: Checkout Repo
+      uses: actions/checkout@v4
+      with:
+        ref: ${{ inputs.ref }}
+
+    - name: Configure VAEC AWS Credentials
+      uses: aws-actions/configure-aws-credentials@v4.0.2
+      with:
+        aws-access-key-id: ${{ inputs.aws-access-key-id }}
```

Comment:
Is this really okay to be passed around? Do we have any risk of this being exposed?

Reply:
Was curious so I poked around and [Github appears to know](https://github.com/department-of-veterans-affairs/notification-api/actions/runs/11409387804/job/31749514923#step:3:11) that this is secret, even though it came from an input. So it's more than just a string, it must keep metadata which is a relief.

Reply:
Yeah, I should have been clearer in the PR. It does mask those values and this is actually convention, as shown in GHA docs: https://docs.github.com/en/actions/sharing-automations/reusing-workflows#using-inputs-and-secrets-in-a-reusable-workflow